runtime/platform_isolate_manager.cc

```diff
@@ -0,0 +1,63 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/runtime/platform_isolate_manager.h"
+
+#include "flutter/runtime/dart_isolate.h"
+
+namespace flutter {
+
+bool PlatformIsolateManager::RegisterPlatformIsolate(Dart_Isolate isolate) {
+  if (is_shutdown_) {
```

Comment:
We do this really rarely, so we don't have to optimize via atomics before and after taking the lock, consider simplifying this code.